benchmark-ph1ex3/neutronics/convert.py

```diff
@@ -0,0 +1,2493 @@
+        kappa.append(kappai)
+    vi = 1/8 * np.ones(8)
+    homogenize_fuel(XS, SP0, kappa, vi, base)
+    print('Layer 8 done')
+
+    # F1 layer 9
+    base = directory + '/mhtgr_F1l9_'
+    XS = []
+    SP0 = []
+    kappa = []
+    for index in range(177, 181):
+        XSi, SP0i, kappai = getfuelxs('xsfiles/fuel171-180.xs', index)
+        XS.append(XSi)
+        SP0.append(SP0i)
+        kappa.append(kappai)
+    for index in range(181, 183):
+        XSi, SP0i, kappai = getfuelxs('xsfiles/fuel181-190.xs', index)
+        XS.append(XSi)
+        SP0.append(SP0i)
+        kappa.append(kappai)
+    vi = 1/6 * np.ones(6)
+    homogenize_fuel(XS, SP0, kappa, vi, base)
+
+    # F2 layer 9
+    base = directory + '/mhtgr_F2l9_'
+    XS = []
+    SP0 = []
+    kappa = []
+    for index in range(183, 191):
+        XSi, SP0i, kappai = getfuelxs('xsfiles/fuel181-190.xs', index)
+        XS.append(XSi)
+        SP0.append(SP0i)
+        kappa.append(kappai)
+    vi = 1/8 * np.ones(8)
+    homogenize_fuel(XS, SP0, kappa, vi, base)
+
+    # F3 layer 9
+    base = directory + '/mhtgr_F3l9_'
+    XS = []
+    SP0 = []
+    kappa = []
+    for index in range(191, 199):
+        XSi, SP0i, kappai = getfuelxs('xsfiles/fuel191-200.xs', index)
+        XS.append(XSi)
+        SP0.append(SP0i)
+        kappa.append(kappai)
+    vi = 1/8 * np.ones(8)
+    homogenize_fuel(XS, SP0, kappa, vi, base)
+    print('Layer 9 done')
+
+    # F1 layer 10
+    base = directory + '/mhtgr_F1l10_'
+    XS = []
+    SP0 = []
+    kappa = []
+    for index in range(199, 201):
+        XSi, SP0i, kappai = getfuelxs('xsfiles/fuel191-200.xs', index)
+        XS.append(XSi)
+        SP0.append(SP0i)
+        kappa.append(kappai)
+    for index in range(201, 205):
+        XSi, SP0i, kappai = getfuelxs('xsfiles/fuel201-210.xs', index)
+        XS.append(XSi)
+        SP0.append(SP0i)
+        kappa.append(kappai)
+    vi = 1/6 * np.ones(6)
+    homogenize_fuel(XS, SP0, kappa, vi, base)
+
+    # F2 layer 10
+    base = directory + '/mhtgr_F2l10_'
+    XS = []
+    SP0 = []
+    kappa = []
+    for index in range(205, 211):
+        XSi, SP0i, kappai = getfuelxs('xsfiles/fuel201-210.xs', index)
+        XS.append(XSi)
+        SP0.append(SP0i)
+        kappa.append(kappai)
+    for index in range(211, 213):
+        XSi, SP0i, kappai = getfuelxs('xsfiles/fuel211-220.xs', index)
+        XS.append(XSi)
+        SP0.append(SP0i)
+        kappa.append(kappai)
+    vi = 1/8 * np.ones(8)
+    homogenize_fuel(XS, SP0, kappa, vi, base)
+
+    # F3 layer 10
+    base = directory + '/mhtgr_F3l10_'
+    XS = []
+    SP0 = []
+    kappa = []
+    for index in range(213, 221):
+        XSi, SP0i, kappai = getfuelxs('xsfiles/fuel211-220.xs', index)
+        XS.append(XSi)
+        SP0.append(SP0i)
+        kappa.append(kappai)
+    vi = 1/8 * np.ones(8)
+    homogenize_fuel(XS, SP0, kappa, vi, base)
+    print('Layer 10 done')
+
+
+def homogenize_collapse_reflec(XS, SP0, vi, base, lim):
```

Comment:
It seems like there is a lot of repeated code between this and ``homog_fuel``, is there a way to reduce the amount of code you're repeating? Perhaps a single function like:
```python
def homogenize(arguments, homogenize_component = 'fuel'):
"""
homogenize_component : string
    Specifies which part of the reactor is being homogenized.
"""
```
Then you can use a "switch" statement to change the behavior for a particular component.